source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst

```diff
@@ -420,11 +420,248 @@ or via command line arguments):
    $ dd if=/dev/mtd6 of=tmp_read.txt bs=num count=1 # read to num bytes to flash
    $ diff tmp_read.txt tmp_write.txt # should be NULL
 
+.. note::
+
+   Make sure UBIFS filesystem is enabled in the kernel (refer to `this
+   section <#enabling-qspi-driver-configurations>`__ for more information).
+
+.. linux_ubifs_start
+
+.. rubric:: Unsorted Block Image File System (UBIFS)
+   :name: linux-ubifs
+
+UBIFS is a flash file system for unmanaged flash memory devices. UBIFS works on
+top of an UBI layer, which is itself on top of MTD (Memory Technology Device)
+layer. UBI + UBIFS takes care of wear leveling, bad-block management, etc.
+
 .. rubric:: Using UBIFS on flash
    :name: using-ubifs-on-flash
 
-Make sure UBIFS filesystem is enabled in the kernel (refer to `this
-section <#enabling-qspi-driver-configurations>`__ for more information).
+.. rubric:: Required Software for UBI image creation
+   :name: linux-required-software-ubifs
+
+Building a UBI File System requires two applications, :command:`ubinize` and
+:command:`mkfs.ubifs`. Both are both provided by ``mtd-utils`` package.
+
+.. code-block:: console
+
+   $ sudo apt-get install mtd-utils
+
+.. rubric:: Building a UBI File System image
+   :name: linux-building-ubi-file-system
+
+Step 1: Create a directory containing the files and directories for the file
+system. It is important that the directory size is smaller than the ``rootfs``
```

Comment:
Provide an example of a base filesystem you can use here. Try base image (provide full names) for nands and tiny for nor or something along those lines